rfcs/0131-sri-hashes.md

```diff
@@ -0,0 +1,50 @@
+---
+feature: sri-hashes
+start-date: 2022-07-25
+author: Winter
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Use SRI hashes and the `hash` parameter for fetchers.
```

Comment:
"SRI hash" needs a reference link with a definition and globally this RFC needs some more context to be explained.

FWIW, since I became aware that this kind of hash was preferred, I have tried to use them more systematically, but I'm annoyed by `nix-prefetch-url` not generating this type of hash (at least all the way to Nix 2.12) and not knowing how to convert a hash generated by `nix-prefetch-url` into an SRI hash.

Reply:
FYI, for conversion I'm using
```
nix hash to-sri --type sha256 theHash
```

Reply:
Thanks! But another thing that is becoming annoying to me lately is to be forced to activate the `nix-command` experimental feature for more and more basic things that I cannot avoid.

Another solution that I've found meanwhile was `openssl dgst -sha256 -binary /path/to/file | openssl base64 -A`, but it's not as convenient.

Reply:
> But another thing that is becoming annoying to me lately is to be forced to activate the `nix-command` experimental feature for more and more basic things that I cannot avoid.

I would rather view it as a missing feature of `nix-hash` that needs to be implemented to get this RFC merged. In other words, we should have

```
nix-hash --sri --type sha256 path/to/file
```

and

```
nix-hash --to-sri --type sha256 theHash
```

Reply:
Thanks! And thanks to you, I also learned that, in the meantime, I can use `nix --extra-experimental-features nix-command file to-sri --type sha256 ...`. I didn't realize that I don't need to activate the experimental feature globally, and this is already better than nothing.